File: statistic.py
#!/usr/bin/env python3

def swap(ages, i, j):
	ages[i] = ages[i] + ages[j]
	ages[j] = ages[i] - ages[j]
	ages[i] = ages[i] - ages[j]

def bubble_sort(ages):
	total_people = len(ages)
	for i in range(total_people):
		for j in range(total_people):
			if(ages[i] > ages[j]):
				swap(ages, i, j)

def f_median(ages):
	total_people = len(ages)

	bubble_sort(ages)

	half = total_people // 2

	if(total_people % 2 != 0):
		return ages[half]
	else:
		half = half - 1
		return (ages[half] + ages[half + 1]) // 2

total_people = int(input("Total de pessoas: "))
age = int(input("Idade da pessoa 1: "))

ages = []
median = age
ages.append(age)
age_accumulator = age
age_of_the_oldest_person = age
age_of_the_youngest_person = age

for counter in range(2, total_people + 1):
	age = int(input("Idade da pessoa {}: ".format(counter)))

	if(age < age_of_the_youngest_person):
		age_of_the_youngest_person = age

	if(age > age_of_the_oldest_person):
		age_of_the_oldest_person = age

	ages.append(age)
	age_accumulator = age_accumulator + age

median = f_median(ages)
mean = age_accumulator // total_people

print("Media das idades:", mean)
print("Mediana das idades:", median)
print("Idade da pessoa mais velha:", age_of_the_oldest_person)
print("Idade da pessoa mais nova:", age_of_the_youngest_person)

File: Statistic.java
class Statistic {
	public static void swap(int ages[], int i, int j) {
		ages[i] = ages[i] + ages[j];
		ages[j] = ages[i] - ages[j];
		ages[i] = ages[i] - ages[j];
	}

	public static void bubbleSort(int ages[]) {
		int totalPeople = ages.length;

		for(int i = 0; i < totalPeople; i++) {
		    for(int j = 0; j < totalPeople; j++) {
		        if(ages[i] > ages[j]) {
		            swap(ages, i, j);
		        }
		    }
		}
	}

	public static int FMedian(int ages[]) {
		int half, totalPeople = ages.length;

		bubbleSort(ages);

		half = totalPeople / 2;

		if(totalPeople % 2 != 0) {
			return ages[half];
		} else {
			half--;
			return (ages[half] + ages[half + 1]) / 2;
		}
	}

	public static void main(String[] args) {
		int[] ages = null;
		int ageOfTheOldestPerson, ageOfTheYoungestPerson;
		int half, totalPeople, ageAccumulator, mean, median, age;

		System.out.print("Total de pessoas: ");
		totalPeople = Integer.parseInt(System.console().readLine());

		System.out.print("Idade da pessoa 1: ");
		age = Integer.parseInt(System.console().readLine());

		median = age;
		ageAccumulator = age;
		ageOfTheOldestPerson = age;
		ageOfTheYoungestPerson = age;
		ages = new int[totalPeople];
		ages[0] = age;

		for(int i = 1; i < totalPeople; i++) {
			System.out.printf("Idade da pessoa %d: ", i + 1);
			age = Integer.parseInt(System.console().readLine());

			if(age < ageOfTheYoungestPerson) {
				ageOfTheYoungestPerson = age;
			}

			if(age > ageOfTheOldestPerson) {
				ageOfTheOldestPerson = age;
			}

			ageAccumulator += age;
			ages[i] = age;
		}

		median = FMedian(ages);
		mean = ageAccumulator / totalPeople;

		System.out.println("Media das idades: " + mean);
		System.out.println("Mediana das idades: " + median);
		System.out.println("Idade da pessoa mais velha: " + ageOfTheOldestPerson);
		System.out.println("Idade da pessoa mais nova: " + ageOfTheYoungestPerson);
	}
}
